test/connectTimeout.js

```diff
@@ -0,0 +1,97 @@
+var Koa = require('koa');
+var agent = require('supertest').agent;
+var proxy = require('../');
+var proxyTarget = require('./support/proxyTarget');
+
+describe('honors connectTimeout option', function() {
+  'use strict';
+
+  var other, http;
+  beforeEach(function() {
+    http = new Koa();
+    other = proxyTarget(8080, 1000, [{
+      method: 'get',
+      path: '/',
+      fn: function(_, res) { res.sendStatus(200); }
+    }]);
+  });
+
+  afterEach(function() {
+    other.close();
+  });
+
+  function assertSuccess(server, done) {
+    agent(server.callback())
+      .get('/')
+      .expect(200)
+      .end(function(err) {
+        if (err) {
+          return done(err);
+        }
+        done();
+      });
+  }
+
+  function assertConnectionTimeout(server, time, done) {
+    agent(server.callback())
+      .get('/')
+      .expect(504)
+      .expect('X-Timout-Reason', 'koa-better-http-proxy timed out your request after ' + time + 'ms.')
+      .end(function(err) {
+        if (err) {
+          return done(err);
+        }
+        done();
+      });
+  }
+
+  describe('when connectTimeout option is set lower than server connect time', function() {
+    it('should fail with CONNECTION TIMEOUT', function(done) {
+      http.use(proxy('http://127.0.0.0', {
```

Comment:
I wanted to find a seemingly more "legit" way to make this happen, but I couldn't and this seems to work 🤷‍♂️